### PR TITLE
Set category on question creation

### LINKED ISSUE #45: Set category on question creation
Currently categories can only be set when a question has already been created. When creating a question it should also be possible to set a category. This should show the same dropdown as used when setting a question category, and it should default to No Category.

### PR DESCRIPTION
**Pull request by tomdohnal**
This adds the possibility to set question categories on question creation. Closes #45.

### CODE REVIEW

File: src/app/components/NewLikertQuestion/index.tsx [hunk 1]
```diff
@@ -82,8 +87,12 @@ class NewLikertQuestionInner extends React.Component<IProps, IState> {
       saving: true,
     });
     this.props.addLikertQuestion(this.props.QuestionSetID, this.state.newQuestion, lv, rv, this.state.leftLabel, this.state.rightLabel, this.state.description)
-    .then(() => {
+    .then((QuestionSet) => {
       this.logQuestionCreatedGAEvent();
+      if (this.state.category) {
+        const QuestionID = QuestionSet.questions.pop().id;
+        this.props.setCategory(this.props.QuestionSetID, QuestionID, this.state.category);
```
Comment (drimpact) [inline src/app/components/NewLikertQuestion/index.tsx]: Sorry I didn't realise I had not exposed category setting on the `AddLikertQuestion` mutation. Let me add that to the mutation tomorrow and we can get rid of this extra jump. See impactasaurus/server#45

File: src/app/components/NewLikertQuestion/index.tsx [hunk 2]
```diff
@@ -145,7 +176,10 @@ class NewLikertQuestionInner extends React.Component<IProps, IState> {
           <Message.Header>New Likert Question</Message.Header>
           <div className="new-likert-form">
             <div className="section upper">
-              <Input className="full" autoFocus type="text" placeholder="Question" onChange={this.setNewQuestion} />
+              <Input className="full question-name" autoFocus type="text" placeholder="Question Name" onChange={this.setNewQuestion} />
```
Comment (drimpact) [inline src/app/components/NewLikertQuestion/index.tsx]: Can we keep the placeholder just "Question"? I think that would make more sense to users.

Comment (tomdohnal) [reply src/app/components/NewLikertQuestion/index.tsx]: Of course, you are right

File: src/app/components/NewLikertQuestion/index.tsx (hunk 1 above)
Comment (tomdohnal) [reply src/app/components/NewLikertQuestion/index.tsx]: Okay, it's going to be much cleaner then. Thanks!

Comment (drimpact) [reply src/app/components/NewLikertQuestion/index.tsx]: This capability has been added and is live

File: src/app/components/NewLikertQuestion/index.tsx [hunk 3]
```diff
@@ -1,19 +1,23 @@
 import * as React from 'react';
-import { Button, ButtonProps, Message, Input } from 'semantic-ui-react';
+import { Button, ButtonProps, Message, Select, Input } from 'semantic-ui-react';
 import {IQuestionMutation, addLikertQuestion} from 'apollo/modules/questions';
+import {IOutcomeResult, getOutcomeSet} from 'apollo/modules/outcomeSets';
+import {ICategoryMutation, setCategory} from 'apollo/modules/categories';
```
Comment (drimpact) [inline src/app/components/NewLikertQuestion/index.tsx]: am I right in thinking that both of these are no longer used?

**Pull request merged**